Source/CarthageKit/Cartfile.swift

```diff
@@ -165,3 +173,43 @@ extension ResolvedCartfile: CustomStringConvertible {
 			.joined(separator: "")
 	}
 }
+
+
+extension String {
+	
+	/// Returns self without any potential trailing Cartfile comment. A Cartfile
+	/// comment starts with the first `commentIndicator` that is not embedded in any quote
+	var strippingTrailingCartfileComment: String {
+		
+		// since Cartfile syntax doesn't support nested quotes, such as `"version-\"alpha\""`
+		// I can just consider any odd-number occurence of a quote as a quote-start, and any
```

Comment:
Nit: grammar: end the `such as` phrase in the `since` line with a comma.

Nit: `I can just consider any` → `simply consider` to remove first-person.